docs/conceptual/node/extrinsics.md

```diff
@@ -2,12 +2,91 @@
+spam prevention mechanism.
+
+Inherents represent data that, in an opinionated fashion, describes one of a number of valid pieces 
+of information. They are assumed to be "true" simply because a sufficiently large number of 
+validators have agreed on them being reasonable.
+
+For example, the author of the block may insert a timestamp inherent into the block. There is no 
+way to prove that a timestamp is true the way the desire to send funds is proved with a signature. 
+Rather, validators accept or reject the block based on how reasonable the other validators find the 
+timestamp, which may mean it is within some acceptable range of their own system clocks.
+
+- [Inherents Reference](https://crates.parity.io/sp_inherents/index.html)
+
+## Signed Transactions
+
+Signed transactions contain a signature of the account that issued the transaction and stands to 
+pay a fee to have the transaction included on chain. Because the value of including signed 
+transactions on-chain can be recognized prior to execution, they can be gossiped on the network 
+between nodes with a low risk of spam.
+
+Signed transactions fit the concept of a transaction in Ethereum or Bitcoin.
+
+## Unsigned Transactions
+
+Some cases call for unsigned transactions. Use unsigned transactions with care, as their validation 
+logic can be difficult.
+
+Since the transaction is not signed, there is nobody to pay a fee. Because of this, the transaction 
+queue lacks economic logic to prevent spam. Unsigned transactions also lack a nonce, making replay 
+protection difficult. A few transactions warrant using the unsigned variant, but they will require 
+some form of spam prevention.
+
+An example of unsigned transactions in Substrate is the 
+[I'm Online](https://github.com/paritytech/substrate/blob/master/frame/im-online/src/lib.rs) 
+heartbeat transaction sent by authorities. The transaction includes a signature from a Session key, 
+which does not control funds and therefore cannot pay a fee. The transaction pool controls spam 
+based on an implementation of `validate_unsigned` that checks if a heartbeat has already been 
+submitted in the session.
+
+## Signed Extension
+
+`SignedExtension` is a trait by which a transaction can be extended with additional data or logic. 
+Signed extensions are used anywhere you want some information about a transaction prior to 
+execution. This is heavily used in the transaction queue.
+
+The runtime can use some of this data, for example the `Call` that will be dispatched, to calculate 
+transaction fees. Signed extensions also include an `AdditionalSigned` type that can hold any 
+encodable data, and therefore allow you to perform any custom logic prior to including or 
+dispatching a transaction. The transaction queue regularly calls functions from `SignedExtension` 
+to validate transactions prior to block construction to avoid including transactions that will fail 
+in blocks.
```

Comment:
```suggestion
in blocks.

Despite the name `SignedExtension` can also be used to verify `UnsignedTransactions`. The `*_unsigned` set of methods can be implemented to encapsulate validation, spam & replay protection logic necessary by the transaction pool.
```

We don't have examples of that yet though, but we plan to rewrite `ImOnline` to actually use `SignedExtension` for validation instead of `UnsignedValidator`.